mne/utils/check.py

```diff
@@ -301,13 +301,15 @@ def _check_preload(inst, msg):
         pass
     else:
         name = "epochs" if isinstance(inst, BaseEpochs) else "raw"
-        if not inst.preload:
+        if not getattr(inst, "preload", False):
             raise RuntimeError(
                 "By default, MNE does not load data into main memory to "
                 "conserve resources. " + msg + " requires %s data to be "
                 "loaded. Use preload=True (or string) in the constructor or "
                 "%s.load_data()." % (name, name)
             )
+        elif name == "epochs" and len(inst) == 0:
```

Comment:
@larsoner The usage from epochs seems to be little compared to the other instances that share this method. And I don't see a case where we would check preload and accept empty epochs. Do you think an extra-arg is necessary?

Reply:
You could try without adding the new kwarg and see if it works